src/zenml/utils/materializer_utils.py

```diff
@@ -0,0 +1,103 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Util functions for models and materializers."""
+
+import os
+import tempfile
+from typing import Any
+
+from ml_metadata.proto.metadata_store_pb2 import Artifact
+
+from zenml.constants import MODEL_METADATA_YAML_FILE_NAME
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.utils import source_utils
+from zenml.utils.yaml_utils import read_yaml, write_yaml
+
+logger = get_logger(__name__)
+
+
+def save_model_metadata(model_artifact: Artifact) -> str:
+    """Save a zenml model artifact metadata to a YAML file.
+
+    This function is used to extract and save information from a zenml model artifact
+    such as the model type and materializer. The extracted information will be
+    the key to loading the model into memory in the inference environment.
+
+    datatype: the model type. This is the path to the model class.
+    materializer: the materializer class. This is the path to the materializer class.
+
+    Args:
+        model_artifact: the artifact to extract the metadata from.
+
+    Returns:
+        The path to the temporary file where the model metadata is saved
+    """
+    metadata = dict()
+    metadata["datatype"] = model_artifact.properties["datatype"].string_value
```

Comment:
These `datatype` and `materializer` string constants are used quite a lot, probably makes sense to have them as a constant somewhere?